ourprojects/models.py

```diff
@@ -38,12 +78,100 @@ class Meta(Record.Meta, TracksRun.Meta, TracksUpdates.Meta):
     """A unique abbreviation."""
     url: str | None = URLField(max_length=255, null=True, default=None)
     """A URL to view."""
+    persons: Person = models.ManyToManyField(Person, related_name="project_persons")
+    """Persons associated with this project."""
+    references: Reference = models.ManyToManyField(
+        "Reference", related_name="project_references"
+    )
+    """References associated with this project."""
     artifacts: Artifact = models.ManyToManyField(
         Artifact, through="ArtifactProject", related_name="Projects"
     )
     """Artifacts labeled with this Project."""
 
 
+class Reference(Record, CanCurate, TracksRun, TracksUpdates, ValidateFields):
+    """References such as a publication or document, with unique identifiers and metadata.
+
+    Example:
+        >>> reference = Reference(
+        ...     name="A Paper Title",
+        ...     abbr="APT",
+        ...     url="https://doi.org/10.1000/xyz123",
+        ...     pubmed_id=12345678,
+        ...     doi="10.1000/xyz123",
+        ...     preprint=False,
+        ...     journal="Nature Biotechnology",
+        ...     description="A groundbreaking research paper.",
+        ...     text="A really informative abstract.",
+        ...     published_at=date(2023, 11, 21),
+        ... ).save()
+    """
+
+    class Meta(Record.Meta, TracksRun.Meta, TracksUpdates.Meta):
+        abstract = False
+
+    id: int = models.AutoField(primary_key=True)
+    """Internal id, valid only in one DB instance."""
+    uid: str = CharField(unique=True, max_length=12, default=ids.base62_12)
+    """Universal id, valid across DB instances."""
+    """Universal id, valid across DB instances."""
+    name: str = CharField(db_index=True)
+    """Title or name of the reference document."""
+    abbr: str | None = CharField(
+        max_length=32,
+        db_index=True,
+        unique=True,
+        null=True,
+    )
+    """A unique abbreviation for the reference."""
+    url: str | None = URLField(null=True, blank=True)
+    """URL linking to the reference."""
+    pubmed_id: int | None = BigIntegerField(null=True)
+    """A PudMmed ID."""
+    doi: int | None = CharField(
+        null=True,
+        db_index=True,
+        validators=[
+            RegexValidator(
+                regex=r"^(?:https?://(?:dx\.)?doi\.org/|doi:|DOI:)?10\.\d+/.*$",
+                message="Must be a DOI (e.g., 10.1000/xyz123 or https://doi.org/10.1000/xyz123)",
+            )
+        ],
+    )
+    """Digital Object Identifier (DOI) for the reference."""
+    preprint: bool = BooleanField(default=False)
```

Comment:
db_index=True

Reply:
I think it's a frequent query, because you want to know whether data is production quality, and from preprints are often less good.